react/src/components/search-app/SearchBarComponent.jsx

```diff
@@ -0,0 +1,126 @@
+import Autocomplete from 'react-autocomplete';
+import React from 'react';
+
+
+export default class SearchBarComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: props.query,
```

Comment:
I've been thinking about it, and I'm not sure if lifting it up is really a good idea. Because then, every component, that uses the SearchBarComponent, needs to handle the `query` property what does not make sense. The SearchBarComponent should be able to work on its own. Maybe, we could just call the prop `initQuery` instead (what is exactly that what it currently is)? (@martomi)